### ISSUE TITLE
Loop Dependent select2 dropdowns update value but do not trigger change() event

### ISSUE THREAD

**Issue by Sijan117** [ermouth/jQuery.my#32]
Tried with: 

select2 4.0.3 (regular + full)
select2 3.5.1 (regular + full)

manifest as follows: 

```javascript
var manifest = {

    data : {id: 18},

    init: function($form) {

        $form.html(this.HTML);

        return factory.getRates()
               .then( response =>{
                   this.RATES = response;
                   console.table(this.RATES);
               } );

    },

    ui: {
        "#uk": {
            init: function ($component) {
                $component.html(this.RATES
                    .reduce((prev, next) => prev + '<option value="' + next.id + '">' + next.uk + '</option>', ''));
                $component.select2({
                    placeholder: "UK"
                });
            },
            bind: function(data, value) {
                    console.log("UK ->  Data:  "+data.toString()+ "    this.data: "+ this.data.toString() +"   Value: "+value); //Debugging
                    var dm = data.id;
                    if(value !== null) dm = value;
                    return dm;
                },
            recalc: "#eu, #usa",
            recalcDepth: 1
        },
        "#eu": {
            init: function ($component) {
                $component.html(this.RATES
                    .reduce((prev, next) => prev + '<option value="' + next.id + '">' + next.eu + '</option>', ''));
                $component.select2({
                    placeholder: "EU"
                });
            },
            bind: function(data, value) {
                    console.log("EU ->  Data:  "+data.toString()+ "    this.data:   "+ this.data.toString() +"   Value: "+value); //Debugging
                    var dm = data.id;
                    if(value !== null) dm = value;
                    return dm;
            },
            recalc: "#usa, #uk",
            recalcDepth: 1
        },
        "#usa": {
            init: function ($component) {
                $component.html(this.RATES
                    .reduce((prev, next) => prev + '<option value="' + next.id + '">' + next.usa + '</option>', ''));
                $component.select2({
                    placeholder: "USA"
                });
            },
            bind: function(data, value) {
                    console.log("USA ->  Data:  "+data.toString()+ "    this.data:   "+ this.data.toString() +"   Value: "+value); //Debugging
                    var dm = data.id;
                    if(value !== null) dm = value;
                    return dm;
            },
            recalc: "#uk, #eu",
            recalcDepth: 1
        }
    },

    HTML:[
        `<div class="row">
            <div>
                <select id="uk"></select>
            </div>
            <div>
                <select id="eu"></select>
            </div>
            <div>
                <select id="usa"></select>
            </div>
         </div>`
        ],

    RATES: []

    };
```

I can see that the values of each dropdown is updated correctly since when I click on it the correct value is already pre-selected, however the currently displayed value on the _other two_ dropdowns does not change. 

If I try to fire a _change_ event myself, the values change however the app halts in an infinite loop of recalc cycles.

**Comment by ermouth** [comment]
You bind functions never store received data to data.id. You should write in this way:
```javascript
function(data, value) {
  if(value != null) data.id = value;
  return data.id;
}
```

**Comment by Sijan117** [comment]
Still getting the same result, values are updated, and the right value is highlighted in the selection box when I click on the selector, but the views of the selection boxes remain the same.

**Comment by ermouth** [comment]
I run code below at jquerymy.com, replacing Bezier code demo. All works pretty fine (I hardcoded RATES somehow) in env of jQuery 2.x.

http://jquerymy.com/index2.html has jQuery 3.x, and code also works pretty fine.

The only difference I see is that I have jquery migrate.

```javascript
({
	data : {id: 18},
	init: function($form) { 	$form.html(this.HTML); },
	ui: {
		"#uk": {
			init: function ($component) {
				$component.html(this.RATES
				.reduce((prev, next) => prev + '<option value="' + next.id + '">' + next.uk + '</option>', ''));
				$component.select2({
					placeholder: "UK"
				});
			},
			bind: function(data, value) {
				if(value != null) data.id = value;
				return data.id;
			},
			recalc: "#eu, #usa",
			recalcDepth: 1
		},
		"#eu": {
			init: function ($component) {
				$component.html(this.RATES
				.reduce((prev, next) => prev + '<option value="' + next.id + '">' + next.eu + '</option>', ''));
				$component.select2({
					placeholder: "EU"
				});
			},
			bind: function(data, value) {
				if(value != null) data.id = value;
				return data.id;
			},
			recalc: "#usa, #uk",
			recalcDepth: 1
		},
		"#usa": {
			init: function ($component) {
				$component.html(this.RATES
				.reduce((prev, next) => prev + '<option value="' + next.id + '">' + next.usa + '</option>', ''));
				$component.select2({
					placeholder: "USA"
				});
			},
			bind: function(data, value) {
				if(value != null) data.id = value;
				return data.id;
			},
			recalc: "#uk, #eu",
			recalcDepth: 1
		}
	},
    HTML:[
        `<div class="row">
            <div>
                <select id="uk"></select>
            </div>
            <div>
                <select id="eu"></select>
            </div>
            <div>
                <select id="usa"></select>
            </div>
         </div>`
        ],

	RATES: [
		{id:18,eu:11,usa:12,uk:13},
		{id:70,eu:71,usa:72,uk:73},
		{id:30,eu:31,usa:32,uk:33}
	]
})
```

**Comment by ermouth** [comment]
Ah, please use latest Select2 of 3.x branch https://github.com/select2/select2/releases/tag/3.5.4. 

Branch 4.x is still bit unstable, so I have not adapt $.my for it yet.

**Comment by Sijan117** [comment]
Yep, that did the trick! Unfortunately the latest version available through npm is v3.5.1, but seems to work great. 

Apologies for not having posted the RATES variable before. 

Cheers.
